src/Components/Authorization/AuthForm.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react"
+
+import ForgotPasswordForm from "./ForgotPasswordForm"
+import LoginForm from "./LoginForm"
+import RegisterForm from "./RegisterForm"
+import { FormComponentType, FormFields, Mode } from "./Types"
+
+type Validator = (val: string) => string
```

Comment:
👍Still thinking through the actual API but depending on the experience it may make sense to support multiple return value types as such 
`(val: string) => string | Promise<string>`

that way we can also support both sync & async validation. A use case is validating emails or looking up if the account already exists